extractors/mongodb/extractor_test.go

```diff
@@ -0,0 +1,178 @@
+	Comments string `bson:"comments,omitempty"`
+}
+
+func TestExtract(t *testing.T) {
+	t.Run("should return error if no user_id in config", func(t *testing.T) {
+		extractor := new(mongodb.Extractor)
+		_, err := extractor.Extract(map[string]interface{}{})
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("should return error if no password in config", func(t *testing.T) {
+		extractor := new(mongodb.Extractor)
+		_, err := extractor.Extract(map[string]interface{}{
+			"user_id": "Gaurav_Ubuntu",
+		})
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("should return error if no host in config", func(t *testing.T) {
+		extractor := new(mongodb.Extractor)
+		_, err := extractor.Extract(map[string]interface{}{
+			"user_id":  "Gaurav_Ubuntu",
+			"password": "abcd",
+		})
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("should return error when testing without mongo running", func(t *testing.T) {
+		extractor := new(mongodb.Extractor)
+		uri := "mongodb://user:abcd@localhost:27017"
+		clientOptions := options.Client().ApplyURI(uri)
+		assertCheck := 0
+		err := MockDataGenerator(clientOptions)
+		if err != nil {
+			assertCheck = 1
+		}
+		_, err = extractor.Extract(map[string]interface{}{
+			"user_id":  "user",
+			"password": "abcd",
+			"host":     "localhost:27017",
+		})
+		if assertCheck == 1 {
+			assert.NotNil(t, err)
+		} else {
+			assert.Nil(t, err)
+		}
+	})
+}
+
+func MockDataGenerator(clientOptions *options.ClientOptions) (err error) {
+	client, err := mongo.NewClient(clientOptions)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	ctx, _ := context.WithTimeout(context.Background(), 2*time.Second)
+	err = client.Connect(ctx)
+	if err != nil {
+		return
+	}
+	err = InsertBlogs(ctx, client)
+	if err != nil {
+		return
+	}
+	err = InsertConnections(ctx, client)
+	if err != nil {
+		return
+	}
+	err = InsertReach(ctx, client)
+	if err != nil {
+		return
+	}
+	client.Disconnect(ctx)
+	return
+}
+
+func InsertBlogs(ctx context.Context, client *mongo.Client) (err error) {
+	collection := client.Database("MeteorMongoExtractorTest").Collection("posts")
+	_, insertErr := collection.InsertMany(ctx, posts)
+	if insertErr != nil {
+		return insertErr
+	}
+	cur, currErr := collection.Find(ctx, bson.D{})
+	if currErr != nil {
+		return currErr
+	}
+	defer cur.Close(ctx)
+	var postsDB []Post
+	if err = cur.All(ctx, &postsDB); err != nil {
+		return err
+	}
+	return
+}
+
+func InsertConnections(ctx context.Context, client *mongo.Client) (err error) {
+	collection := client.Database("MeteorMongoExtractorTest").Collection("connection")
+	_, insertErr := collection.InsertMany(ctx, connections)
+	if insertErr != nil {
+		return insertErr
+	}
+	cur, currErr := collection.Find(ctx, bson.D{})
+	if currErr != nil {
+		return currErr
+	}
+	defer cur.Close(ctx)
+	var connectionsDB []Connection
+	if err = cur.All(ctx, &connectionsDB); err != nil {
+		return err
+	}
+	return
+}
+
+func InsertReach(ctx context.Context, client *mongo.Client) (err error) {
```

Comment:
all these functions could be change to private if we are not exposing it elsewhere.